### PR TITLE
genSym symbols rendered as b1_416049 instead of b1416049

### PR DESCRIPTION
**Pull request by timotheecour**
(a little bit similar in spirit to https://github.com/nim-lang/Nim/pull/11939 @Clyybber )
helps with debugging when looking at rendered code (via `repr` etc)
the lack of separator makes it hard to guess what symbol a gensym'ed symbol refers to especially when the symbol ends with a number. This PR makes it clear.
Note this is only a rendering thing, and should not affect semantics.

```nim
import macros
macro fun(): untyped =
  result = quote do:
    template bar() =
      var b = 1
      var b1 = 2
      var b2 = 3
  echo repr(result)
fun()
```

before PR:
```nim
template bar() =
  var b424045 = 1
  var b1424046 = 2
  var b2424047 = 3
```

after PR:
```nim
template bar() =
  var b_416045 = 1
  var b1_416046 = 2
  var b2_416047 = 3
```

### CODE REVIEW

File: compiler/renderer.nim [hunk 1]
```diff
@@ -839,12 +841,12 @@ proc gident(g: var TSrcGen, n: PNode) =
...
-  put(g, t, s, if n.kind == nkSym and renderSyms in g.flags: n.sym else: nil)
+  var suffix = ""
   if n.kind == nkSym and (renderIds in g.flags or sfGenSym in n.sym.flags or n.sym.kind == skTemp):
+    suffix = genSymSuffix(n.sym)
     when defined(debugMagics):
-      put(g, tkIntLit, $n.sym.id & $n.sym.magic)
-    else:
-      put(g, tkIntLit, $n.sym.id)
+      suffix.add "_" & $n.sym.magic
+  put(g, t, s & suffix, if n.kind == nkSym and renderSyms in g.flags: n.sym else: nil)
```
Comment (Araq) [inline compiler/renderer.nim]: The old way didn't do a string concat for every rendered identifier.

**Comment by Araq** [comment]
If you rewrite the logic, ensure it's not obviously slower than before. Plus the old version was just as simple to read.

File: compiler/renderer.nim (hunk 1 above)
Comment (timotheecour) [reply compiler/renderer.nim]: done

**Comment by timotheecour** [comment]
PTAL

**Pull request merged**